lib/code_corps/analytics/segment.ex

```diff
@@ -63,14 +61,65 @@ defmodule CodeCorps.Analytics.Segment do
   def track(conn, :signed_up) do
     conn |> do_track("Signed Up")
   end
-  def track(conn, _event) do
+  def track(%Plug.Conn{} = conn, _event) do
     conn # return conn without event to track
   end
 
+  # switch to tracking in the context of ja_resource
+
+  @actions_without_properties [:updated_profile, :signed_in, :signed_out, :signed_up]
+
+  def track({:ok, record}, action, %Plug.Conn{} = conn) when action in @actions_without_properties do
```

Comment:
Is there any way to pattern match on the `record`'s type at all for extra type safety?